web3.js/test/connection.test.ts

```diff
@@ -3756,6 +3756,53 @@ describe('Connection', function () {
     verifySignatureStatus(response, expectedErr);
   });
 
+  if (mockServer) {
+    it('returnData on simulateTransaction', async () => {
+      const tx = new Transaction();
+      tx.feePayer = Keypair.generate().publicKey;
```

Comment:
@joncinque It's worth noting that this will only work with a funded feePayer, which makes simulation somewhat problematic for second-order return-data apis. That doesn't need to hold up anything here, but just something to think about for usage and documentation purposes.

Reply:
Yeah true, I was thinking we could at least support it in its current state and improve on it later.  Just to be totally clear, when you say "that doesn't need to hold up anything here", do you mean that you're ok with the approach?

Also, did we ever come up with a model for simulating transactions without a funded fee payer?

Reply:
Yep, I’m fine with this approach.
No, we haven’t designed implemented a way to bypass fee-payer check in simulation